src/main/scala/is/hail/methods/MinHash.scala

```diff
@@ -0,0 +1,64 @@
+package is.hail.methods
+
+import breeze.linalg.{DenseMatrix, DenseVector, min}
+import is.hail.variant.{Genotype, Variant, VariantDataset}
+import is.hail.annotations.Annotation
+import scala.collection.mutable.HashSet
+import scala.util.hashing.{MurmurHash3 => MH3}
+import scala.util.Random
+
+object MinHash {
+  // returns a k by vds.nSamples matrix
+  def kMinHash(vds: VariantDataset, k: Int): DenseMatrix[Int] = {
+    val n = vds.nSamples
+    def seqOp(l: DenseMatrix[Int], r: (Variant, (Annotation, Iterable[Genotype]))): DenseMatrix[Int] = r match {
+      case (_, (_, gs)) =>
+        val gsItr = gs.hardCallGenotypeIterator
+        val hashes = DenseVector.fill[Int](k)(Random.nextInt())
+        var i = 0
+        while(i < n) {
+          //TODO: Write a min UFunc that modifies the left arg in place
+          if (gsItr.next() > 0) l(::,i) := min(l(::,i), hashes)
+          i += 1
+        }
+        l
+    }
+    val zeroValue = DenseMatrix.fill[Int](k, n)(Int.MaxValue)
+    vds.rdd.aggregate(zeroValue)(seqOp, min(_,_))
+  }
+
+  //mat is a k-by-n matrix, which will be split into r-by-n submatrices for Min-LSH
+  def findSimilarPairs(mat: DenseMatrix[Int], r: Int): Set[(Int, Int)] =
+    findSimilarPairs(mat, r, mat.rows / r)
```

Comment:
It does still work. I set it up this way to make benchmarking easier: so we can just do one `minHash` pass with a large `k`, then run `findSimilarPairs` with various parameters on the same `minHash` output. Though in retrospect, we could just as easily pass submatrices of the `minHash` output to `findSimilarPairs`; maybe that would be clearer?